[Actionable Observability] - add alertDetailAppSection to the APM Rule Details page

fkanout opened the pull request:
## Summary

It fixes #145572 and it fixes #143183 by
- Adding an abstraction layer for the Timeseries chart and using it without APM's context and hooks
- Create and link `AlertDetailAppSection` new component to all APM rule types
- Provide the new Alert Details page under Observability using APM charts

<img width="1205" alt="Screenshot 2022-11-08 at 17 22 17" src="https://user-images.githubusercontent.com/6838659/200619684-0f9eaf5b-4618-4dd3-8539-f4bc13107c2e.png">

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)


## For the reviewers
- Enable the feature flag `xpack.observability.unsafe.alertDetails.apm.enabled: true` in `kibana.yml`.
- Create an APM rule that fires alerts
- Click on `View alert details` from the Alert table

fkanout committed:
WIP

fkanout committed:
Fix type issue

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
Update alert_details.tsx

fkanout committed:
Fully cleaned and working Latency chart

fkanout committed:
Remove unused code

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

fkanout committed:
Update comparisonEnabled to false

fkanout committed:
Merge branch 'main'

Conflicts:
	x-pack/plugins/apm/public/components/alerting/register_apm_alerts.ts

fkanout committed:
abstract the timeseries chart

fkanout committed:
update folder structure

fkanout committed:
share types

fkanout committed:
Add throughput chart

fkanout committed:
Add hte transaction rate failed

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
Add panel and chart titles

fkanout committed:
Add p95 and p99 for the chart

fkanout committed:
Fix type

fkanout committed:
Update tech fields

fkanout committed:
Use APM hook for transaction type and add the app section for all APM rule types

fkanout committed:
Add spacer

fkanout committed:
Add comment for getOrRedirectToTransactionType

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
Add time range calculation

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
Update the comment

fkanout committed:
Update snapshot and fix tests

fkanout committed:
Fix tests

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
Update service field to be array:false

elasticmachine commented:
Pinging @elastic/actionable-observability (Team: Actionable Observability)

fkanout committed:
Fix tests

elasticmachine commented:
Pinging @elastic/apm-ui (Team:APM)

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
Fix tests

maryam-saeidi commented on x-pack/plugins/rule_registry/common/assets/field_maps/technical_rule_field_map.test.ts:
```diff
@@ -43,9 +43,6 @@ it('matches snapshot', () => {
       "kibana.alert.end": Object {
         "type": "date",
       },
-      "kibana.alert.flapping": Object {
```
It's not used anymore?

ymao1 replied:
This was just added [recently by the framework](https://github.com/elastic/kibana/pull/143920) so we'd like to keep it :)

fkanout committed:
Fix test and update snapshot

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
Fix test that rely on the rule registry

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
fix rangeTo by using the ALERT_END

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

MiriamAparicio commented:
hi @fkanout, I’m reviewing the PR, I’m manually testing it but seems that the charts are not getting any data, plus getting a 404 when requesting` GET internal/alerting/rule` and the requests for latency, failed transactions rate, and throughput seems that are not made, as they don’t appear in the dev tools, I'm testing in my oblt-cluster in edge
![image](https://user-images.githubusercontent.com/31922082/201629886-9e6dc338-942c-404e-9594-74817390b17e.png)

maryam-saeidi commented:
I've tested locally and the graphs on the alert details page match what I see in the APM app. 👏🏻 

@fkanout Any idea about the rule fetch error in the console?
![image](https://user-images.githubusercontent.com/12370520/201632855-741c8e6f-c489-4647-9a4d-6e9ba3ff36bf.png)

fkanout committed:
Fix the 404 error when fetch the rule

maryam-saeidi commented:
> @maryam-saeidi, thanks for catching that error while fetching the rule. I just pushed a [fix](https://github.com/elastic/kibana/pull/143298/commits/1692b206eba933ac371a6c50d6c9b238a6ea943d) for it. Can you retest it, please?

I see you made `ruleId` optional, in which scenario we will not have a `ruleId` for an alert? Seems to me we should always have it.

dgieselaar commented on x-pack/plugins/rule_registry/common/assets/field_maps/technical_rule_field_map.ts:
```diff
@@ -189,6 +189,16 @@ export const technicalRuleFieldMap = {
     array: false,
     required: false,
   },
+  [Fields.SERVICE_NAME]: {
+    type: 'keyword',
+    array: false,
+    required: false,
+  },
+  [Fields.TRANSACTION_TYPE]: {
```
These are not technical fields, they are APM-specific metadata.

XavierM replied:
Dario, you are right! Should we just add a new file `apm_rule_field_map.ts` in this folder for now `x-pack/plugins/rule_registry/common/assets/field_maps` and we can add a new function `parseApmFields` and add it to the function [parseAlert](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/public/pages/alerts/components/parse_alert.ts#L36)

Or do you APM folks already owned a package where we can put these new files?

ymao1 replied:
They should be added to the `initializeIndex` function when APM rules are registered with the rule registry.

XavierM replied:
That will be good but it might be hard for them to get that done before FF, maybe they can create an issue to fix it

ymao1 replied:
Xavier explained the context of this to me so I am ok with his proposal

dgieselaar commented on x-pack/plugins/apm/public/context/apm_service/apm_service_context.tsx:
```diff
@@ -97,7 +97,7 @@ export function getOrRedirectToTransactionType({
   transactionTypes: string[];
   agentName?: string;
-  history: History;
+  history?: History;
```
I don't think we should make this optional, because it's easy to forget. If you need only the current transaction type, you'll probably want to split up the function (e.g. you'd create `getTransactionType` that doesn't execute a redirect and use that).

sqren replied:
I agree. Let's rename it to `getTransactionType` and inline the `replace` call in `ApmServiceContextProvider`:

```ts
  const transactionType = getTransactionType({
    transactionType: query.transactionType,
    transactionTypes,
    agentName,
  });

  // Replace transactionType in the URL in case it is not one of the types returned by the API
  replace(history, { query: { transactionType } });
```

fkanout replied:
done

dgieselaar commented on x-pack/plugins/apm/public/components/alerting/rule_types/register_apm_rule_types.ts:
```diff
@@ -33,7 +33,7 @@ export function registerApmRuleTypes(
         reason: fields[ALERT_REASON]!,
         link: getAlertUrlErrorCount(
-          String(fields[SERVICE_NAME][0]!),
+          String(fields[SERVICE_NAME]![0]),
```
Why did this change?

sqren replied:
As mentioned elsewhere, it would be awesome to have proper types for alert and rule params.

fkanout replied:
@dgieselaar, because the service name field is not required. This will be added as TODO now, and fixed in a separate PR when we move these APM specific fields to `initializeIndex`

dgieselaar commented on x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section.tsx:
```diff
@@ -0,0 +1,510 @@
+  const latencyAggregationType = getAggsTypeFromRule(
+    params.aggregationType as string
+  );
+  const alertDurationMS =
+    alert && alert.fields[ALERT_DURATION]
+      ? // duration is us, convert it to MS
+        (alert.fields[ALERT_DURATION] as unknown as number) / 1000
+      : 0;
```
Why is there a fallback here? Is `ALERT_DURATION` ever nullish? If it's just a type issue, either the type probably should be fixed, or if that's too complicated, use the nullish coalescing operator (`!`).

dgieselaar commented on x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section.tsx:
```diff
@@ -0,0 +1,510 @@
+  if (ruleAggType === '99th') return LatencyAggregationType.p99;
+  return LatencyAggregationType.avg;
+};
+export function AlertDetailsAppSectionTransactionDuration({
```
IMHO this component would very much benefit from being broken up into several appropriately named functions. It is quite long and hard to grok.

fkanout replied:
Agree. I renamed the component, cleaned it up, and exported types and helpers

dgieselaar commented on x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section.tsx:
```diff
@@ -0,0 +1,510 @@
+  timeZone,
+}: AlertDetailsAppSectionProps) {
+  const params = rule.params;
+  const environment = String(params.environment);
```
I'm not sure if this gives you the behaviour that you expect. `params.environment` should either be a non-empty string, `ENVIRONMENT_ALL` or `ENVIRONMENT_NOT_DEFINED` when passing it as a query parameter to an API endpoint and when using it as a URL parameter for the UI. if `params.environment` can be undefined, it will be an empty string which is not valid. I would suggest falling back to `ENVIRONMENT_ALL` in that case (we have constants for this).

fkanout replied:
@dgieselaar you are right. Done

dgieselaar commented on x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section.tsx:
```diff
@@ -0,0 +1,510 @@
+  const offset = '1d';
+  const ruleWindowSizeMS = moment
+    .duration(
+      rule.params.windowSize as unknown as number,
+      rule.params.windowUnit as unknown as any
```
why is this needed?

fkanout replied:
We need this to calculate `TWENTY_TIMES_RULE_WINDOW_MS`

dgieselaar commented on x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section.tsx:
```diff
@@ -0,0 +1,510 @@
+        (alert.fields[ALERT_DURATION] as unknown as number) / 1000
+      : 0;
+  const serviceName = String(alert.fields[SERVICE_NAME]);
+  const comparisonEnabled = false;
```
This is a kibana advanced setting. See `getComparisonEnabled`.

fkanout replied:
same as https://github.com/elastic/kibana/pull/143298#discussion_r1021814564

dgieselaar commented on x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section.tsx:
```diff
@@ -0,0 +1,510 @@
+      : 0;
+  const serviceName = String(alert.fields[SERVICE_NAME]);
+  const comparisonEnabled = false;
+  const offset = '1d';
```
I don't think that this is the right default value. We probably need to calculate this based on the given time range.

fkanout replied:
If I got the code correctly, the `offset` value is ONLY considered when the `comparisonEnabled` is `true`.
Currently, we don't do any comparison, so I'm adding it here as it's required by the charts' props. And I pulled all the related code for  `comparisonEnabled ` and `offset` from the query. 

Is that ok @dgieselaar ?

dgieselaar reviewed (commented):
Some early feedback. I haven't finished looking through it all. It would be helpful if you can split up`alert_details_app_section.tsx` in several files, because it's quite long and a little hard to review.

fkanout committed:
Code review

fkanout committed:
Update component name

fkanout committed:
Update component file

maryam-saeidi commented on x-pack/plugins/observability/public/pages/alert_details/components/alert_details.tsx:
```diff
@@ -27,19 +31,31 @@ import { paths } from '../../../config/paths';
   const CasesContext = getCasesContext();
   const userCasesPermissions = canUseCases();
+  const { rule } = useFetchRule({
+    ruleId: alert?.fields[ALERT_RULE_UUID],
```
Shouldn't we fetch the rule when we have alert fields (and as a result the ruleId)?
Then we don't need to make ruleId parameter in useFetchRule optional.

fkanout replied:
@maryam-saeidi, yeah there is a custom hooks dependency here.

fkanout committed:
Update apm service context

fkanout commented:
>I agree. Let's rename it to getTransactionType and inline the replace call in ApmServiceContextProvider.

@sqren, good point. I moved the `replace` call to the body of `ApmServiceContextProvider.` 
However, I added a condition to call it, to match the intended behavior . **i.e., call it only when transaction type is not one of the types returned by the API**

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
code review use isNill from _

fkanout commented on x-pack/plugins/apm/public/components/shared/charts/latency_chart/index.tsx:
```diff
@@ -38,7 +38,7 @@ const options: Array<{ value: LatencyAggregationType; text: string }> = [
-function filterNil<T>(value: T | null | undefined): value is T {
+export function filterNil<T>(value: T | null | undefined): value is T {
   return value != null;
 }
```
Done

fkanout commented on x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section/types.ts:
```diff
@@ -0,0 +1,15 @@
+import { TopAlert } from '@kbn/observability-plugin/public/pages/alerts';
+
+export interface AlertDetailsAppSectionProps {
+  rule: Rule<RuleTypeParams>;
```
Sure, I updated it as suggested.

fkanout committed:
code review - update type

CoenWarmer commented on x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section/alert_details_app_section.tsx:
```diff
@@ -175,7 +176,8 @@ export function AlertDetailsAppSection({
     currentPeriod,
     comparisonEnabled && isTimeComparison(offset) ? previousPeriod : undefined,
-  ].filter(filterNil);
+  ].filter((timeserie) => !isNil(timeserie)) as Array<APMChartSpec<Coordinate>>;
```
Ah, noticed this is by @sqren's [request](https://github.com/elastic/kibana/pull/143298#discussion_r1022685963).

sqren replied:
Yes, my bad! If `isNil` looses type information we should keep our own `filterNil` implementation.

fkanout replied:
@sqren @dgieselaar both functions `compact` and `isNil` will lose types. I'm reverting to `filterNil`

fkanout commented on x-pack/plugins/apm/public/components/shared/charts/timeseries_chart.tsx:
```diff
@@ -91,30 +64,22 @@ export function TimeseriesChart({
+}: TimeseriesChartProps) {
   const history = useHistory();
-  const { core } = useApmPluginContext();
   const { annotations } = useAnnotationsContext();
   const { chartRef, updatePointerEvent } = useChartPointerEventContext();
   const theme = useTheme();
   const chartTheme = useChartTheme();
```
The abstraction work done in this PR is to make the charts usable from outside the APM context, aka not tightly coupled. 
There is a second round of work for annotations, where we could do more abstraction, but it will be in another PR.

fkanout commented on x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section/types.ts:
```diff
@@ -0,0 +1,15 @@
+
+export interface AlertDetailsAppSectionProps {
+  rule: Rule<RuleTypeParams>;
+  alert: TopAlert;
```
@sqren  `alert` is already typed. maybe I'm missing something? the Alert and Rule are shared types

fkanout replied:
`Rule` params are typed as you suggested

sqren replied:
`TopAlert` doesn't have the right types for `alert.fields`. If they had we wouldn't have to do:

```
alert.fields[ALERT_DURATION] as unknown as number
```

Same with `rule`:

```
rule.params.windowSize as unknown as number,
rule.params.windowUnit as unknown as any
```

fkanout committed:
switch back to filterNil

fkanout committed:
Add TODOs for SERVICE_NAME

fkanout committed:
Add TODO

fkanout commented on packages/kbn-rule-data-utils/src/technical_field_names.ts:
```diff
@@ -19,6 +19,8 @@ const EVENT_ACTION = 'event.action' as const;
 const EVENT_MODULE = 'event.module' as const;
 const SPACE_IDS = `${KIBANA_NAMESPACE}.space_ids` as const;
+const SERVICE_NAME = 'service.name' as const;
+const TRANSACTION_TYPE = 'transaction.type' as const;
```
Done

fkanout replied:
@dgieselaar, It seems straightforward to me! What do you think @XavierM?

fkanout committed:
Fix test for getTransactionType

fkanout commented:
@dgieselaar @sqren Thank you for your review and comments!
I addressed the points you mentioned.
One thing left for a follow-up PR is the APM-related fields. I have created an issue for it https://github.com/elastic/kibana/issues/145572, and will work with @XavierM on it.

Please note this PR is only the beginning effort to deliver the Alert Details page. More PRs to follow up. e.g., for annotations 

If you think this PR is now functional and clean enough to be merged, please give it approval.

maryam-saeidi reviewed (approved):
Great effort! 💪🏻 

AO changes LGTM

sorenlouv reviewed (approved):
lgtm. Thanks for doing this! Seeing this is scheduled for 8.7 I'm not concerned about the changes to APM (we now have time to address any issues that may pop up).

@dgieselaar any final thoughts?

dgieselaar commented:
@sqren having another look

dgieselaar commented on x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section/alert_details_app_section.tsx:
```diff
@@ -0,0 +1,464 @@
+  const latencyMaxY = getMaxY(timeseriesLatency);
+  const latencyFormatter = getDurationFormatter(latencyMaxY);
+  const preferredAnomalyTimeseriesLatency =
+    usePreferredServiceAnomalyTimeseries(ApmMlDetectorType.txLatency);
```
This doesn't work without the appropriate context. I don't think we need to use it here, or we need to wrap this component with the context providers.

fkanout replied:
I didn't know that; I was aiming to keep the style of charts aligned on both pages(APM and Observability). 
After deleting them, the style is still the same, which is perfect!

dgieselaar commented on x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section/alert_details_app_section.tsx:
```diff
@@ -0,0 +1,464 @@
+
+                  <EuiFlexItem grow={false}>
+                    <EuiIconTip
+                      content={i18n.translate('xpack.apm.errorRate.tip', {
+                        defaultMessage:
+                          "The percentage of failed transactions for the selected service. HTTP server transactions with a 4xx status code (client error) aren't considered failures because the caller, not the server, caused the failure.",
+                      })}
```
should this be a shared label? I assume we use it in another place too? This shouldn't pass CI either, because the label id is duplicated.

fkanout replied:
I exported it from _original_ failed_transaction_rate_chart. I didn't find a shared file where APM puts the shared i18n

dgieselaar commented on x-pack/plugins/apm/public/context/apm_service/apm_service_context.tsx:
```diff
@@ -62,13 +61,20 @@ export function ApmServiceContextProvider({
-    history,
   });
 
+  // Replace transactionType in the URL in case it is not one of the types returned by the API
+  if (
+    currentTransactionType &&
+    currentTransactionType !== query.transactionType
+  ) {
+    replace(history, { query: { transactionType: currentTransactionType } });
+  }
```
Can you put this back into the function? We also have tests for this. I'm assuming they break if you do this.

fkanout replied:
Yeah, I'm aware that there are tests for it. I'm updating the tests to ensure this update has no regression.

dgieselaar replied:
ok! Can you make sure you're testing the redirect logic still?

fkanout replied:
@dgieselaar, in case it's become overcropped for this PR, would you agree to put back the history as an optional argument?

dgieselaar replied:
I'm sorry, but I don't follow this. AFAIR the suggestion was for you to extract the part of the function that returns the transaction type, and import that extracted function. That would allow you to keep the test as-is and not making any other changes to a critical part of our application code. Is there anything I'm not seeing that is stopping you from doing that? To clarify, you would have:

- `getOrRedirectToTransactionType`, which is the function we have today
- `getTransactionType`, that only returns the transaction type based on its input, and is called by `getOrRedirectTransactionType`
- you would import `getTransactionType` in your code

sqren replied:
... I see your point about the ability to test it when it's encapsulated in a function.

dgieselaar replied:
Ah I did not see your follow up comment on my suggestion. Apologies for the confusion @fkanout. Still, I think it's critical that we have unit tests for `getOrRedirectToTransactionType`, so I suggest we stick to my initial suggestion (as described above).

fkanout replied:
@dgieselaar, it was a bit confusing, but no worries! I applied the suggested implementation

dgieselaar commented on x-pack/plugins/apm/public/context/apm_service/apm_service_context.test.tsx:
```diff
@@ -15,23 +15,21 @@ describe('getOrRedirectToTransactionType', () => {
-          history,
         })
       ).toBe('custom');
       expect(history.replace).not.toHaveBeenCalled();
```
This will fail if you remove the `replace` call from the hook (obviously). It's important that we export a separate function that does the redirect logic so we can more easily test it.

dgieselaar reviewed (changes requested):
@fkanout There's still quite some stuff to address IMHO.

fkanout committed:
code review - coexists getTransactionType and getOrRedirectToTransactionType

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
code review - remove casting after defining types

fkanout committed:
code review - ALERT_DURATION type forcing

fkanout committed:
code review - errorRateI18n

fkanout commented on x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section/alert_details_app_section.tsx:
```diff
@@ -0,0 +1,464 @@
+      }),
+    // It should only update when the data has changed
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [data]
```
I copied it as it's from use_transaction_latency_chart_fetcher.ts. I put `latencyAggregationType` in the dependencies hook

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
Merge branch '143183-alertDetailAppSection-apm' of github.com:fkanout/kibana into 143183-alertDetailAppSection-apm

fkanout committed:
code review - remove timeseries unused style

fkanout committed:
Use TAdditionalMetaFields for service name and transaction type

fkanout committed:
fix tests and update the snapshot

fkanout commented:
@dgieselaar, done as proposed here https://github.com/elastic/kibana/pull/143298#discussion_r1026335777

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

dgieselaar commented on x-pack/plugins/apm/public/components/alerting/ui_components/alert_details_app_section/alert_details_app_section.tsx:
```diff
@@ -0,0 +1,464 @@
+  const preferredAnomalyTimeseriesLatency =
+    usePreferredServiceAnomalyTimeseries(ApmMlDetectorType.txLatency);
+
+  /* Latency Chart */
+
+  /* Throughput Chart */
```
@fkanout did you miss this one?

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

CoenWarmer commented on x-pack/plugins/observability/public/pages/alert_details/components/alert_details.tsx:
```diff
@@ -27,19 +31,31 @@ import { paths } from '../../../config/paths';
 
+  useEffect(() => {
+    if (alert) {
+      setRuleTypeModel(ruleTypeRegistry.get(alert?.fields[ALERT_RULE_TYPE_ID]!));
```
You might not need `useEffect` and `useState` in this instance.

```
const ruleTypeModel = alert ? ruleTypeRegistry.get(alert?.fields[ALERT_RULE_TYPE_ID]) : undefined
```

Would also work.

CoenWarmer replied:
Also `ruleTypeRegistry` [will throw an error if the passed rule type can't be found](https://github.com/elastic/kibana/blob/main/x-pack/plugins/triggers_actions_ui/public/application/type_registry.ts:L47), so if you want to be safe it might be better to do something like:

```
const getRuleTypeModel = (ruleTypeId: string | undefined): RuleTypeModel<RuleTypeParams> | undefined => {
    try {
      return ruleTypeRegistry.get(alert?.fields[ALERT_RULE_TYPE_ID]);
    } catch (error) {
      return undefined
    }
}

const ruleTypeModel = getRuleTypeModel(alert?.fields[ALERT_RULE_TYPE_ID]);
```

fkanout replied:
@CoenWarmer will check that, and see if this is the behavior that we would like to have.
Regarding the suggestion for the `getRuleTypeModel ` function, it doesn't have access to the Alert object as it's outside the component. I'm not sure about exposing all these arguments.

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout committed:
Fix cypress tests

fkanout committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

fkanout commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into 143183-alertDetailAppSection-apm

dgieselaar reviewed (approved):
Thanks for getting this over the finish line @fkanout!

fkanout committed:
Simplify the check
